### PR TITLE
Use more accurate spans when proposing adding lifetime to item

### PR DESCRIPTION
**Pull request by estebank**

### CODE REVIEW

File: compiler/rustc_resolve/src/late/diagnostics.rs [hunk 1]
```diff
@@ -2076,16 +2076,40 @@ impl<'tcx> LifetimeContext<'_, 'tcx> {
                 for param in params {
                     if let Ok(snippet) = self.tcx.sess.source_map().span_to_snippet(param.span) {
                         if snippet.starts_with('&') && !snippet.starts_with("&'") {
-                            introduce_suggestion
-                                .push((param.span, format!("&'a {}", &snippet[1..])));
-                        } else if let Some(stripped) = snippet.strip_prefix("&'_ ") {
```
Comment (oli-obk) [inline compiler/rustc_resolve/src/late/diagnostics.rs:2081]: there's a lot of repetition in this diff. maybe we should create a `(String, Span)` newtype with functions like `strip_prefix` or `slice` that operate on both at the same time?

Comment (estebank) [reply compiler/rustc_resolve/src/late/diagnostics.rs:2081]: Done

**Comment by estebank** [comment]
@oli-obk just pushed a quick fix for a double space issue in the newtype code.

@bors r=oli-obk

**Pull request merged**